### PR TITLE
Add UI to create a invite

### PR DESCRIPTION
**Pull request by cquiroz**
This is a first baby step to create invites to the UI
We should test this and make improvements based on feedback

https://github.com/gemini-hlsw/explore/assets/3615303/616e1655-7a80-468d-a6ff-29e64d4d88d1

### CODE REVIEW

**Review by hugo-vrijswijk — COMMENTED** [review]
LGTM. Should we also display the actual link next to the copy button?

File: explore/src/main/scala/explore/proposal/InviteUserPopup.scala [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                )(^.autoComplete := "off")
+              ),
+              <.div(LucumaPrimeStyles.FormColumn)(
+                <.label(
+                  "Invite ready, please send the link below to you collaborator, it won't be displayed again"
+                )
+              ).when(key.when(_.isDefined)),
+              key.get.map(key =>
+                <.div(LucumaPrimeStyles.FormColumn)(
+                  CopyControl("Invite Link", dom.window.location.origin + s"/reedem-invite/${key}")
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply explore/src/main/scala/explore/proposal/InviteUserPopup.scala:102]: Uh, thanks for noticing it

File: explore/src/main/scala/explore/proposal/ProgramUsers.scala [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            Button(
+              severity = Button.Severity.Secondary,
+              size = Button.Size.Small,
+              icon = Icons.UserPlus,
+              tooltip = "Create CoI invitation",
+              onClickE = props.ref.toggle
+            ).tiny.compact
+          ),
+          ProgramUsersTable(props.users),
+          "Pending invitations".when(props.invitations.nonEmpty),
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply explore/src/main/scala/explore/proposal/ProgramUsers.scala:94]: Do you mean for multiple blocks with the same conditions? 👍

File: model/shared/src/main/scala/explore/model/UserInvitation.scala [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import cats.Eq
+import cats.derived.*
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.cats.given
+import eu.timepit.refined.string.MatchesRegex
+import io.circe.Decoder
+
+// Matchtes the odb standard for email validation
+type EmailPred =
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply model/shared/src/main/scala/explore/model/UserInvitation.scala:14]: Yes, we will move this to core and should put some tests there

File: explore/src/clue/scala/queries/common/InvitationQueriesGQL.scala [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+import clue.GraphQLOperation
+import clue.annotation.GraphQL
+import lucuma.schemas.ObservationDB
+
+object InvitationQueriesGQL:
+  @GraphQL
+  trait CreateInviteMutation extends GraphQLOperation[ObservationDB]:
+    val document: String = s"""
+      mutation inviteToProgram($$programId: ProgramId!, $$recipientEmail: String!) {
```
Comment (toddburnside) [inline explore/src/clue/scala/queries/common/InvitationQueriesGQL.scala:14]: Does the name `inviteToProgram` do anything here. I see we only "name" the mutations here and in the user preferences queries. It doesn't seem as if the name is ever used, though.

Comment (cquiroz) [reply explore/src/clue/scala/queries/common/InvitationQueriesGQL.scala:14]: Not really, I thought it was required to put a name 🤷

Comment (toddburnside) [reply explore/src/clue/scala/queries/common/InvitationQueriesGQL.scala:14]: And I didn't realize it was possible to put a name. 😆

**Comment by cquiroz** [comment]
The link is a bit large for the space we have
At any rate this is the initial version awaiting user feedback

**Pull request merged**